src/player.js

```diff
@@ -763,6 +844,131 @@ class Player {
         return this.set('color', color);
     }
 
+    /**
+     * A promise to set all colors for the player.
+     *
+     * @promise SetColorsPromise
+     * @fulfill {string[]} The colors were successfully set.
+     * @reject {TypeError} The string was not a valid hex or rgb color.
+     * @reject {ContrastError} The color was set, but the contrast is
+     *         outside of the acceptable range.
+     * @reject {EmbedSettingsError} The owner of the player has chosen to
+     *         use a specific color.
+     */
+    /**
+     * Set the colors of this player to a hex or rgb string. Setting the
+     * color may fail if the owner of the video has set their embed
+     * preferences to force a specific color.
+     * The colors should be passed in as an array: [colorOne, colorTwo, colorThree, colorFour].
+     * If a color should not be set, the index in the array can be left as null.
+     *
+     * @param {string[]} colors Array of the hex or rgb color strings to set.
+     * @return {SetColorsPromise}
+     */
+    setColors(colors = []) {
```

Comment:
Why an array here instead of individual parameters?

Reply:
This was to match the OEmbed approach of using an array, but now that you mention it, it probably would make sense to support an object of fields. I'll add support for that (but we should still support an array to not conflict with the OEmbed approach).

Reply:
I think an array parameter or individual parameters makes more sense than an object parameter.

Reply:
Hmm, I'm thinking if a user only wants to update a single color, it's a bit clunky to have to pass an array with a bunch of commas or nulls in order to do that. But on the flip side, the user would have to know the key name for an object, which would require them doing some unwanted investigating. I guess if we want to keep things as simple as possible for both users and maintainers, sticking with the array would be the way to go - otherwise we might as well keep all the individual getters/setters.